gin/sentrygin.go

```diff
@@ -57,11 +57,11 @@ func (h *handler) handle(c *gin.Context) {
 	options := []sentry.SpanOption{
 		sentry.WithOpName("http.server"),
 		sentry.ContinueFromRequest(c.Request),
-		sentry.WithTransactionSource(sentry.SourceURL),
+		sentry.WithTransactionSource(sentry.SourceRoute),
 	}
 
 	transaction := sentry.StartTransaction(ctx,
-		fmt.Sprintf("%s %s", c.Request.Method, c.Request.URL.Path),
+		fmt.Sprintf("%s %s", c.Request.Method, c.FullPath()),
```

Comment:
```
For not found routes returns an empty string.
```

This causes issues for 404 routes, as the transaction name is now always something like `GET `.
We should check for the return of `c.FullPath()`. If it's empty, use `c.Request.URL.Path` as a fallback and set `sentry.WithTransactionSource(sentry.SourceURL)`.

Reply:
That's cool that you found it! Done
https://github.com/getsentry/sentry-go/pull/675/commits/3b50c09995a5a8f149545f5e9ea4f52d9e77d061

Reply:
@antonsacred I think what @cleptric suggests is more like, do the check and fallback here in `handle` directly (not in `defer`), and set TransactionSource depending on the value of `c.FullPath()`.